src/js/components/Investigation/FindingCard.tsx

```diff
@@ -27,47 +22,55 @@ const StyledMagnifyingGlass = styled(MagnifyingGlass)`
   }
 `
 
-type Props = {finding: Finding}
+type Props = {finding: Finding; connId: string; spaceId: string}
 
-export default React.memo<Props>(function FindingCard({finding}: Props) {
+export default React.memo<Props>(function FindingCard({
+  finding,
+  connId,
+  spaceId
+}: Props) {
   const dispatch = useDispatch()
-  const clusterId = useSelector(Current.getConnectionId)
-  const spaceIds = useSelector(Spaces.ids(clusterId))
-  const findingSpaceName = get(finding, ["search", "spaceName"], "")
 
   function onClick() {
     dispatch(Search.restore(finding.search))
     dispatch(submitSearch({history: true, investigation: false}))
   }
 
-  function renderWarning() {
-    const findingSpaceId = get(finding, ["search", "spaceId"], "")
-    const tip = "This space no longer exists"
-
-    if (includes(spaceIds, findingSpaceId)) return null
-
-    return (
-      <div
-        className="warning-body"
-        data-tip={tip}
-        data-effect="solid"
-        data-place="right"
-      >
-        <Warning />
-        <ReactTooltip />
-      </div>
-    )
-  }
-
   const menu = usePopupMenu([
     {
       label: "Delete",
-      click: () => globalDispatch(Investigation.deleteFindingByTs(finding.ts))
+      click: () =>
+        remote.dialog
+          .showMessageBox({
+            type: "warning",
+            title: "Delete History Entry",
+            message: `Are you sure you want to delete this history entry?`,
+            buttons: ["OK", "Cancel"]
+          })
+          .then(({response}) => {
+            if (response === 0)
+              globalDispatch(
+                Investigation.deleteFindingByTs(connId, spaceId, finding.ts)
+              )
+          })
```

Comment:
I found that the recursive functionality of the "FindingTree" nodes vs the iterative "FindingCard" made consolidation a little tricky. This menu, for example, I had thought to pass in from both components' parent but since the "Delete" functionality is different between each (FindingCard just deletes that one entry, where the NodeRow first finds its children and then deletes the whole branch) I decided to leave as is. I also thought to just pass in the common menu element (Delete All, which functions the same for both) but ultimately thought that keeping each array of menu logic together within the corresponding component and having a single piece of duplication was better than breaking the logic across component boundaries. Lmk if you prefer the latter though!